### ISSUE TITLE
segfault when 3rd param is missing when using Task.async

### ISSUE THREAD

**Issue by apeckham** [bitwalker/timex#39]
I get a segfault when I forgot to include the 3rd parameter (:strftime) of DateFormat.format!

```elixir
defmodule Mix.Tasks.Segfault do
  use Mix.Task
  use Timex

  def run(_args) do
    task = Task.async(fn ->
      Date.from({{2015,1,1},{1,1,1}}) |> DateFormat.format! "%Y-%m-%d" #, :strftime
    end)
    Task.await task, :infinity
  end
end
```

```
{"init terminating in do_boot",{badarg,[{erlang,byte_size,[[{message,<<59 bytes>>}]],[]},{'Elixir.Exception',format_banner,3,[{file,"lib/exception.ex"},{line,106}]},{'Elixir.Exception',format_exit,2,[{file,"lib/exception.ex"},{line,171}]},{'Elixir.Exception',format_banner,3,[{file,"lib/exception.ex"},{line,118}]},{'Elixir.Kernel.CLI',print_error,3,[{file,"lib/kernel/cli.ex"},{line,113}]},{'Elixir.Kernel.CLI',exec_fun,2,[{file,"lib/kernel/cli.ex"},{line,97}]},{'Elixir.Kernel.CLI',run,2,[{file,"lib/kernel/cli.ex"},{line,37}]},{init,start_it,1,[]}]}}
Abort trap: 6
```

On IRC, @ericmj says timex should unwrap [message: message] here: https://github.com/bitwalker/timex/blob/master/lib/dateformat/format_error.ex#L9

Elixir 1.0.3
Erlang/OTP 17 [erts-6.3.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

**Comment by bitwalker** [comment]
Right on, thanks for reporting, I'll take a stab at fixing this up tonight and push out a fix.

**Comment by bitwalker** [comment]
Pushed this fix plus some others in a new release, 0.13.4. Thanks again for the report!
